### PR TITLE
HiFaceGAN merge (autopep8 processed)

### PR DESCRIPTION
**Pull request by Lotayou**
@xinntao 
    当前测试代码跑通，只添加了四个代码文件：网络结构，数据集，配置文件，测试脚本，基本上独立于你的原始工程，只是调用了一些FileClient之类的工具类。可以看下能不能fast-forward，有conflict的话就等你更新了底层结构之后再合并吧。
    其中，模型实现和配置文件放在`basicsr/models/arch`里面。dataset提供了原文的退化配置，后面准备令用户自己根据要研究的内容选择相应的退化配置，直接从FFHQ高清生成低质图像。为兼容我的原版代码，额外添加了一个Options类用于配置参数，并提供了与dict的转换函数（递归的，可以处理多重嵌套），这些放在`basicsr/utils/options.py`里。
    我手头只有一张11G的显卡，暂时无法提供DDP支持和大规模数据预训练模型，可能要等到入职了。

凌波

### CODE REVIEW

**Comment by Lotayou** [comment]
Update 20210429：
Run autopep8 for code-reformating. See if this passes pylint test.

**Comment by Lotayou** [comment]
@xinntao Requesting workflow approval

File: basicsr/models/archs/hifacegan_options.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
+import sys
+from basicsr.utils import Options
+
```
Comment (xinntao) [inline basicsr/models/archs/hifacegan_options.py:3]: I recommend merging this option to a YAML file, so that it could be consistent with the original repo.

File: basicsr/models/archs/sync_batchnorm/batchnorm.py [hunk 2]
```diff
@@ -0,0 +1,416 @@
+# -*- coding: utf-8 -*-
```
Comment (xinntao) [inline basicsr/models/archs/sync_batchnorm/batchnorm.py:1]: Why not use the official syncbn? May try to use the official sync bn

**Comment by xinntao** [comment]
Hi @Lotayou, I have updated the original repo (especially the organization and styles).  So it may introduce some conflicts. 

 The current implementation may have some major 'design conflicts' with the original repo. 

I advise updating and clean it to meet similar styles with the original repo. It may take some time to modify the current implementations, even with retraining.  
Otherwise, this repo will become more and more complicated when more methods are coming.

File: basicsr/models/archs/sync_batchnorm/batchnorm.py (hunk 2 above)
Comment (Lotayou) [reply basicsr/models/archs/sync_batchnorm/batchnorm.py:1]: According to [pytorch doc](https://pytorch.org/docs/stable/generated/torch.nn.SyncBatchNorm.html) SyncBatchNorm only supports DDP training, so I cannot test it now. I shall settle with InstanceNorm2D for now.

File: basicsr/models/archs/hifacegan_options.py (hunk 1 above)
Comment (Lotayou) [reply basicsr/models/archs/hifacegan_options.py:3]: Guess I'll need to change every option call instead:) But let's do it your way.

**Comment by Lotayou** [comment]
@xinntao Agreed. I will create another PR instead. This fork can be archived.
